### ISSUE TITLE
ActiveRecord::RecordNotFound: Couldn't find Post with 'id'=0 probably due to primary key value was nil

### ISSUE THREAD

**Issue by yahonda** [rsim/oracle-enhanced#1318]
### Steps to reproduce
```ruby
begin
  require 'bundler/inline'
rescue LoadError => e
  $stderr.puts 'Bundler version 1.10 or later is required. Please update your Bundler'
  raise e
end

gemfile(true) do
  source 'https://rubygems.org'
  gem "rails", github: "rails/rails"
  gem "arel", github: "rails/arel"
  gem 'activerecord-oracle_enhanced-adapter', github: "rsim/oracle-enhanced", branch: "nextval"
  gem 'ruby-oci8'
  gem 'minitest'
  gem 'byebug'
end

require 'active_record'
require 'minitest/autorun'
require 'logger'
require 'active_record/connection_adapters/oracle_enhanced_adapter'
require 'byebug'

# Ensure backward compatibility with Minitest 4
Minitest::Test = MiniTest::Unit::TestCase unless defined?(Minitest::Test)

# Set Oracle enhanced adapter specific connection parameters
DATABASE_NAME = ENV['DATABASE_NAME'] || 'orcl'
DATABASE_HOST = ENV['DATABASE_HOST']
DATABASE_PORT = ENV['DATABASE_PORT']
DATABASE_USER = ENV['DATABASE_USER'] || 'oracle_enhanced'
DATABASE_PASSWORD = ENV['DATABASE_PASSWORD'] || 'oracle_enhanced'
DATABASE_SYS_PASSWORD = ENV['DATABASE_SYS_PASSWORD'] || 'admin'

CONNECTION_PARAMS = {
  :adapter => "oracle_enhanced",
  :database => DATABASE_NAME,
  :host => DATABASE_HOST,
  :port => DATABASE_PORT,
  :username => DATABASE_USER,
  :password => DATABASE_PASSWORD,
  :nls_numeric_characters => ',.'
}

ActiveRecord::Base.establish_connection(CONNECTION_PARAMS)

ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :posts, :force => true do |t|
    t.integer :price
  end
end

class Post < ActiveRecord::Base
end

class BugTest < Minitest::Test
  def test_post_reloads_based_on_the_pk_value_100
    post = Post.create!(price: 100)
    assert post.price, 100
    post.reload
    assert post.price, 100
  end
end
```

### Expected behavior
This test should pass.

```ruby
begin
  require 'bundler/inline'
rescue LoadError => e
  $stderr.puts 'Bundler version 1.10 or later is required. Please update your Bundler'
  raise e
end

gemfile(true) do
  source 'https://rubygems.org'
  gem "rails", github: "rails/rails"
  gem "arel", github: "rails/arel"
  gem 'activerecord-oracle_enhanced-adapter', github: "rsim/oracle-enhanced", branch: "master"
  gem 'ruby-oci8'
  gem 'minitest'
  gem 'byebug'
end

require 'active_record'
require 'minitest/autorun'
require 'logger'
require 'active_record/connection_adapters/oracle_enhanced_adapter'
require 'byebug'

# Ensure backward compatibility with Minitest 4
Minitest::Test = MiniTest::Unit::TestCase unless defined?(Minitest::Test)

# Set Oracle enhanced adapter specific connection parameters
DATABASE_NAME = ENV['DATABASE_NAME'] || 'orcl'
DATABASE_HOST = ENV['DATABASE_HOST']
DATABASE_PORT = ENV['DATABASE_PORT']
DATABASE_USER = ENV['DATABASE_USER'] || 'oracle_enhanced'
DATABASE_PASSWORD = ENV['DATABASE_PASSWORD'] || 'oracle_enhanced'
DATABASE_SYS_PASSWORD = ENV['DATABASE_SYS_PASSWORD'] || 'admin'

CONNECTION_PARAMS = {
  :adapter => "oracle_enhanced",
  :database => DATABASE_NAME,
  :host => DATABASE_HOST,
  :port => DATABASE_PORT,
  :username => DATABASE_USER,
  :password => DATABASE_PASSWORD,
  :nls_numeric_characters => ',.'
}

ActiveRecord::Base.establish_connection(CONNECTION_PARAMS)

ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :posts, :force => true do |t|
    t.integer :price
  end
end

class Post < ActiveRecord::Base
end

class BugTest < Minitest::Test
  def test_post_reloads_based_on_the_pk_value_100
    post = Post.create!(price: 100)
    assert post.price, 100
    post.reload
    assert post.price, 100
  end
end

[yahonda@scw-bfe314 rep_nextval]$ ruby rep_oracle_current.rb
Fetching git://github.com/rails/rails.git
Fetching git://github.com/rails/arel.git
Fetching git://github.com/rsim/oracle-enhanced.git
Fetching gem metadata from https://rubygems.org/...........
Fetching version metadata from https://rubygems.org/..
Resolving dependencies....
Using rake 12.0.0
Using concurrent-ruby 1.0.5
Using i18n 0.8.1
Using minitest 5.10.1
Using thread_safe 0.3.6
Using builder 3.2.3
Using erubi 1.6.0
Using mini_portile2 2.1.0
Using rack 2.0.1
Using nio4r 2.0.0
Using websocket-extensions 0.1.2
Using mime-types-data 3.2016.0521
Using arel 8.0.0 from git://github.com/rails/arel.git (at master@437aa3a)
Using ruby-plsql 0.6.0
Using bundler 1.14.6
Using byebug 9.0.6
Using method_source 0.8.2
Using thor 0.19.4
Using ruby-oci8 2.2.3
Using tzinfo 1.2.3
Using nokogiri 1.7.1
Using rack-test 0.6.3
Using sprockets 3.7.1
Using websocket-driver 0.6.5
Using mime-types 3.1
Using activesupport 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using loofah 2.0.3
Using mail 2.6.5
Using rails-dom-testing 2.0.2
Using globalid 0.4.0
Using activemodel 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using rails-html-sanitizer 1.0.3
Using activejob 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using activerecord 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using actionview 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using activerecord-oracle_enhanced-adapter 1.9.0.alpha from git://github.com/rsim/oracle-enhanced.git (at master@f51c9c2)
Using actionpack 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using actioncable 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using actionmailer 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using railties 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using sprockets-rails 3.2.0
Using rails 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
-- create_table(:posts, {:force=>true})
D, [2017-05-01T19:43:11.748463 #1813] DEBUG -- :    (91.1ms)  DROP TABLE "POSTS"
D, [2017-05-01T19:43:11.771935 #1813] DEBUG -- :    (22.4ms)  DROP SEQUENCE "POSTS_SEQ"
D, [2017-05-01T19:43:11.809862 #1813] DEBUG -- :    (36.7ms)  CREATE TABLE "POSTS" ("ID" NUMBER(38) NOT NULL PRIMARY KEY, "PRICE" NUMBER(38))
D, [2017-05-01T19:43:11.821360 #1813] DEBUG -- :    (10.8ms)  CREATE SEQUENCE "POSTS_SEQ" START WITH 10000
   -> 0.2663s
D, [2017-05-01T19:43:11.887931 #1813] DEBUG -- :    (4.3ms)  SELECT SYS_CONTEXT('userenv', 'current_schema') FROM dual
D, [2017-05-01T19:43:11.922617 #1813] DEBUG -- :    (15.2ms)  SELECT cols.column_name AS name, cols.data_type AS sql_type, cols.data_default, cols.nullable, cols.virtual_column, cols.hidden_column, cols.data_type_owner AS sql_type_owner, DECODE(cols.data_type, 'NUMBER', data_precision, 'FLOAT', data_precision, 'VARCHAR2', DECODE(char_used, 'C', char_length, data_length), 'RAW', DECODE(char_used, 'C', char_length, data_length), 'CHAR', DECODE(char_used, 'C', char_length, data_length), NULL) AS limit, DECODE(data_type, 'NUMBER', data_scale, NULL) AS scale, comments.comments as column_comment FROM all_tab_cols cols, all_col_comments comments WHERE cols.owner = 'ORACLE_ENHANCED' AND cols.table_name = 'AR_INTERNAL_METADATA' AND cols.hidden_column = 'NO' AND cols.owner = comments.owner AND cols.table_name = comments.table_name AND cols.column_name = comments.column_name ORDER BY cols.column_id
D, [2017-05-01T19:43:11.930596 #1813] DEBUG -- :   ActiveRecord::InternalMetadata Load (3.8ms)  SELECT  "AR_INTERNAL_METADATA".* FROM "AR_INTERNAL_METADATA" WHERE "AR_INTERNAL_METADATA"."KEY" = :a1 FETCH FIRST :a2 ROWS ONLY  [["key", "environment"], ["LIMIT", 1]]
D, [2017-05-01T19:43:11.933741 #1813] DEBUG -- :    (2.1ms)  SELECT SYS_CONTEXT('userenv', 'current_schema') FROM dual
Run options: --seed 28207

# Running:

D, [2017-05-01T19:43:11.983447 #1813] DEBUG -- :    (10.3ms)  SELECT cols.column_name AS name, cols.data_type AS sql_type, cols.data_default, cols.nullable, cols.virtual_column, cols.hidden_column, cols.data_type_owner AS sql_type_owner, DECODE(cols.data_type, 'NUMBER', data_precision, 'FLOAT', data_precision, 'VARCHAR2', DECODE(char_used, 'C', char_length, data_length), 'RAW', DECODE(char_used, 'C', char_length, data_length), 'CHAR', DECODE(char_used, 'C', char_length, data_length), NULL) AS limit, DECODE(data_type, 'NUMBER', data_scale, NULL) AS scale, comments.comments as column_comment FROM all_tab_cols cols, all_col_comments comments WHERE cols.owner = 'ORACLE_ENHANCED' AND cols.table_name = 'POSTS' AND cols.hidden_column = 'NO' AND cols.owner = comments.owner AND cols.table_name = comments.table_name AND cols.column_name = comments.column_name ORDER BY cols.column_id
D, [2017-05-01T19:43:12.229812 #1813] DEBUG -- :    (186.6ms)  SELECT LOWER(view_name) FROM all_views WHERE owner = SYS_CONTEXT('userenv', 'current_schema')
D, [2017-05-01T19:43:12.235155 #1813] DEBUG -- :   Sequence (2.5ms)  select us.sequence_name from all_sequences us where us.sequence_owner = 'ORACLE_ENHANCED' and us.sequence_name = upper('POSTS_seq')
D, [2017-05-01T19:43:12.244286 #1813] DEBUG -- :   Primary Key (8.3ms)  SELECT cc.column_name FROM all_constraints c, all_cons_columns cc WHERE c.owner = 'ORACLE_ENHANCED' AND c.table_name = 'POSTS' AND c.constraint_type = 'P' AND cc.owner = c.owner AND cc.constraint_name = c.constraint_name
D, [2017-05-01T19:43:12.256261 #1813] DEBUG -- :   Sequence (2.0ms)  select us.sequence_name from all_sequences us where us.sequence_owner = 'ORACLE_ENHANCED' and us.sequence_name = upper('POSTS_seq')
D, [2017-05-01T19:43:12.262029 #1813] DEBUG -- :   Primary Key (5.0ms)  SELECT cc.column_name FROM all_constraints c, all_cons_columns cc WHERE c.owner = 'ORACLE_ENHANCED' AND c.table_name = 'POSTS' AND c.constraint_type = 'P' AND cc.owner = c.owner AND cc.constraint_name = c.constraint_name
D, [2017-05-01T19:43:12.266492 #1813] DEBUG -- :   Primary Key Trigger (3.8ms)            SELECT trigger_name
          FROM all_triggers
          WHERE owner = 'ORACLE_ENHANCED'
            AND trigger_name = q'[POSTS_PKT]'
            AND table_owner = 'ORACLE_ENHANCED'
            AND table_name = q'[POSTS]'
            AND status = 'ENABLED'

D, [2017-05-01T19:43:12.297574 #1813] DEBUG -- :   SQL (20.5ms)  INSERT INTO "POSTS" ("PRICE", "ID") VALUES (:a1, :a2)  [["price", 100], ["id", 10000]]
D, [2017-05-01T19:43:12.315310 #1813] DEBUG -- :   Post Load (11.7ms)  SELECT  "POSTS".* FROM "POSTS" WHERE "POSTS"."ID" = :a1 FETCH FIRST :a2 ROWS ONLY  [["id", 10000], ["LIMIT", 1]]
.

Finished in 0.348330s, 2.8708 runs/s, 5.7417 assertions/s.

1 runs, 2 assertions, 0 failures, 0 errors, 0 skips
$
```

### Actual behavior
```ruby
[yahonda@scw-bfe314 rep_nextval]$ ruby rep_oracle_nextval.rb
Fetching git://github.com/rails/rails.git
Fetching git://github.com/rails/arel.git
Fetching git://github.com/rsim/oracle-enhanced.git
Fetching gem metadata from https://rubygems.org/...........
Fetching version metadata from https://rubygems.org/..
Resolving dependencies....
Using rake 12.0.0
Using concurrent-ruby 1.0.5
Using i18n 0.8.1
Using minitest 5.10.1
Using thread_safe 0.3.6
Using builder 3.2.3
Using erubi 1.6.0
Using mini_portile2 2.1.0
Using rack 2.0.1
Using nio4r 2.0.0
Using websocket-extensions 0.1.2
Using mime-types-data 3.2016.0521
Using arel 8.0.0 from git://github.com/rails/arel.git (at master@437aa3a)
Using ruby-plsql 0.6.0
Using bundler 1.14.6
Using byebug 9.0.6
Using method_source 0.8.2
Using thor 0.19.4
Using ruby-oci8 2.2.3
Using tzinfo 1.2.3
Using nokogiri 1.7.1
Using rack-test 0.6.3
Using sprockets 3.7.1
Using websocket-driver 0.6.5
Using mime-types 3.1
Using activesupport 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using loofah 2.0.3
Using mail 2.6.5
Using rails-dom-testing 2.0.2
Using globalid 0.4.0
Using activemodel 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using rails-html-sanitizer 1.0.3
Using activejob 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using activerecord 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using actionview 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using activerecord-oracle_enhanced-adapter 2.0.0.alpha from git://github.com/rsim/oracle-enhanced.git (at nextval@d0bdb06)
Using actionpack 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using actioncable 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using actionmailer 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using railties 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
Using sprockets-rails 3.2.0
Using rails 5.2.0.alpha from git://github.com/rails/rails.git (at master@aaf99e3)
-- create_table(:posts, {:force=>true})
D, [2017-05-01T19:43:49.026861 #2028] DEBUG -- :    (88.2ms)  DROP TABLE "POSTS"
D, [2017-05-01T19:43:49.050324 #2028] DEBUG -- :    (22.4ms)  DROP SEQUENCE "POSTS_SEQ"
D, [2017-05-01T19:43:49.100286 #2028] DEBUG -- :    (48.8ms)  CREATE TABLE "POSTS" ("ID" NUMBER(38) GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY, "PRICE" NUMBER(38))
D, [2017-05-01T19:43:49.109869 #2028] DEBUG -- :    (8.9ms)  CREATE SEQUENCE "POSTS_SEQ" START WITH 10000
   -> 0.2769s
D, [2017-05-01T19:43:49.171078 #2028] DEBUG -- :    (4.3ms)  SELECT SYS_CONTEXT('userenv', 'current_schema') FROM dual
D, [2017-05-01T19:43:49.204445 #2028] DEBUG -- :    (14.9ms)  SELECT cols.column_name AS name, cols.data_type AS sql_type, cols.data_default, cols.nullable, cols.virtual_column, cols.hidden_column, cols.data_type_owner AS sql_type_owner, DECODE(cols.data_type, 'NUMBER', data_precision, 'FLOAT', data_precision, 'VARCHAR2', DECODE(char_used, 'C', char_length, data_length), 'RAW', DECODE(char_used, 'C', char_length, data_length), 'CHAR', DECODE(char_used, 'C', char_length, data_length), NULL) AS limit, DECODE(data_type, 'NUMBER', data_scale, NULL) AS scale, comments.comments as column_comment FROM all_tab_cols cols, all_col_comments comments WHERE cols.owner = 'ORACLE_ENHANCED' AND cols.table_name = 'AR_INTERNAL_METADATA' AND cols.hidden_column = 'NO' AND cols.owner = comments.owner AND cols.table_name = comments.table_name AND cols.column_name = comments.column_name ORDER BY cols.column_id
D, [2017-05-01T19:43:49.211989 #2028] DEBUG -- :   ActiveRecord::InternalMetadata Load (3.6ms)  SELECT  "AR_INTERNAL_METADATA".* FROM "AR_INTERNAL_METADATA" WHERE "AR_INTERNAL_METADATA"."KEY" = :a1 FETCH FIRST :a2 ROWS ONLY  [["key", "environment"], ["LIMIT", 1]]
D, [2017-05-01T19:43:49.214942 #2028] DEBUG -- :    (1.9ms)  SELECT SYS_CONTEXT('userenv', 'current_schema') FROM dual
Run options: --seed 8302

# Running:

D, [2017-05-01T19:43:49.263403 #2028] DEBUG -- :    (10.2ms)  SELECT cols.column_name AS name, cols.data_type AS sql_type, cols.data_default, cols.nullable, cols.virtual_column, cols.hidden_column, cols.data_type_owner AS sql_type_owner, DECODE(cols.data_type, 'NUMBER', data_precision, 'FLOAT', data_precision, 'VARCHAR2', DECODE(char_used, 'C', char_length, data_length), 'RAW', DECODE(char_used, 'C', char_length, data_length), 'CHAR', DECODE(char_used, 'C', char_length, data_length), NULL) AS limit, DECODE(data_type, 'NUMBER', data_scale, NULL) AS scale, comments.comments as column_comment FROM all_tab_cols cols, all_col_comments comments WHERE cols.owner = 'ORACLE_ENHANCED' AND cols.table_name = 'POSTS' AND cols.hidden_column = 'NO' AND cols.owner = comments.owner AND cols.table_name = comments.table_name AND cols.column_name = comments.column_name ORDER BY cols.column_id
D, [2017-05-01T19:43:49.502266 #2028] DEBUG -- :    (185.6ms)  SELECT LOWER(view_name) FROM all_views WHERE owner = SYS_CONTEXT('userenv', 'current_schema')
D, [2017-05-01T19:43:49.507358 #2028] DEBUG -- :   Sequence (2.3ms)  select us.sequence_name from all_sequences us where us.sequence_owner = 'ORACLE_ENHANCED' and us.sequence_name = upper('POSTS_seq')
D, [2017-05-01T19:43:49.516409 #2028] DEBUG -- :   Primary Key (8.3ms)  SELECT cc.column_name FROM all_constraints c, all_cons_columns cc WHERE c.owner = 'ORACLE_ENHANCED' AND c.table_name = 'POSTS' AND c.constraint_type = 'P' AND cc.owner = c.owner AND cc.constraint_name = c.constraint_name
D, [2017-05-01T19:43:49.527905 #2028] DEBUG -- :   Sequence (2.1ms)  select us.sequence_name from all_sequences us where us.sequence_owner = 'ORACLE_ENHANCED' and us.sequence_name = upper('POSTS_seq')
D, [2017-05-01T19:43:49.533514 #2028] DEBUG -- :   Primary Key (4.9ms)  SELECT cc.column_name FROM all_constraints c, all_cons_columns cc WHERE c.owner = 'ORACLE_ENHANCED' AND c.table_name = 'POSTS' AND c.constraint_type = 'P' AND cc.owner = c.owner AND cc.constraint_name = c.constraint_name
D, [2017-05-01T19:43:49.537694 #2028] DEBUG -- :   Primary Key Trigger (3.5ms)            SELECT trigger_name
          FROM all_triggers
          WHERE owner = 'ORACLE_ENHANCED'
            AND trigger_name = q'[POSTS_PKT]'
            AND table_owner = 'ORACLE_ENHANCED'
            AND table_name = q'[POSTS]'
            AND status = 'ENABLED'

D, [2017-05-01T19:43:49.568042 #2028] DEBUG -- :   SQL (20.4ms)  INSERT INTO "POSTS" ("PRICE", "ID") VALUES (:a1, :a2)  [["price", 100], ["id", 10000]]
D, [2017-05-01T19:43:49.585867 #2028] DEBUG -- :   Post Load (11.3ms)  SELECT  "POSTS".* FROM "POSTS" WHERE "POSTS"."ID" = :a1 FETCH FIRST :a2 ROWS ONLY  [["id", 0], ["LIMIT", 1]]
E

Finished in 0.338295s, 2.9560 runs/s, 2.9560 assertions/s.

  1) Error:
BugTest#test_post_reloads_based_on_the_pk_value_100:
ActiveRecord::RecordNotFound: Couldn't find Post with 'id'=0
    /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/rails-aaf99e3cfc70/activerecord/lib/active_record/relation/finder_methods.rb:341:in `raise_record_not_found_exception!'
    /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/rails-aaf99e3cfc70/activerecord/lib/active_record/relation/finder_methods.rb:468:in `find_one'
    /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/rails-aaf99e3cfc70/activerecord/lib/active_record/relation/finder_methods.rb:448:in `find_with_ids'
    /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/rails-aaf99e3cfc70/activerecord/lib/active_record/relation/finder_methods.rb:66:in `find'
    /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/rails-aaf99e3cfc70/activerecord/lib/active_record/querying.rb:3:in `find'
    /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/rails-aaf99e3cfc70/activerecord/lib/active_record/core.rb:171:in `find'
    /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/rails-aaf99e3cfc70/activerecord/lib/active_record/persistence.rb:462:in `block in reload'
    /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/rails-aaf99e3cfc70/activerecord/lib/active_record/scoping/default.rb:35:in `block in unscoped'
    /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/rails-aaf99e3cfc70/activerecord/lib/active_record/relation.rb:337:in `scoping'
    /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/rails-aaf99e3cfc70/activerecord/lib/active_record/scoping/default.rb:35:in `unscoped'
    /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/rails-aaf99e3cfc70/activerecord/lib/active_record/persistence.rb:462:in `reload'
    /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/rails-aaf99e3cfc70/activerecord/lib/active_record/attribute_methods/dirty.rb:50:in `reload'
    /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/rails-aaf99e3cfc70/activerecord/lib/active_record/associations.rb:283:in `reload'
    /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/rails-aaf99e3cfc70/activerecord/lib/active_record/autosave_association.rb:236:in `reload'
    /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/rails-aaf99e3cfc70/activerecord/lib/active_record/aggregations.rb:13:in `reload'
    rep_oracle_nextval.rb:62:in `test_post_reloads_based_on_the_pk_value_100'

1 runs, 1 assertions, 0 failures, 1 errors, 0 skips
$
```

### System configuration
**Rails version**: master branch

**Oracle enhanced adapter version**: nextval branch

**Ruby version**: ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-linux]

**Oracle Database version**:Oracle Database 12c Enterprise Edition Release 12.2.0.1.0 - 64bit Production

**Comment by yahonda** [comment]
It is caused by the default value given to `new_column` method is sequence_name.nextval statement,
not the default value.

```sql
SQL> select data_default from user_tab_columns
  2  where table_name = 'POSTS' and column_name = 'ID'
  3  ;

DATA_DEFAULT
--------------------------------------------------------------------------------
"ORACLE_ENHANCED"."ISEQ$$_103449".nextval

SQL>
```
A dirty workaround is to replace this statement with nil.

**Comment by yahonda** [comment]
Closed by #1340

**Comment by fuzzy76** [comment]
If one were stuck to some legacy code that require Rails 4 and therefore oracle_enhanced-adapter-1.6.9 - is there any way to work around this from the calling code?
